### PR TITLE
Add blogpost on JAX London 2016

### PR DESCRIPTION
**Pull request by yannickdeturck**
Review and suggestions most appreciated!

### CODE REVIEW

File: _posts/2016-11-18-JAX-London-2016.md [hunk 1]
```diff
@@ -0,0 +1,691 @@
...
+- Repositories
+- Aggregates
+
+About [Domain-Driven Design aggregates](http://martinfowler.com/bliki/DDD_Aggregate.html):
+
+- Cluster of objects that can be treated as a unit
+- Graph consisting of a root entity and one or more other entities and value objects
+- Typically business entities are aggregates e.g., Customer, Account, Order, Product
+
+Chris talked about the problems you have to deal with when dealing with microservices and that practicing Domain-Driven Design can help you a lot to address these.
```
Comment (tomvdbulck) [inline _posts/2016-11-18-JAX-London-2016.md]: you have to deal with when dealing with ... lot's of with's.

maybe replace to you have to deal with by you have to handle

File: _posts/2016-11-17-JAX-London-2016.md [hunk 2]
```diff
@@ -0,0 +1,691 @@
...
+- NIST (33)
+- NCSC (europe)(44)
+- Cliff Berg’s set (185)
+
+Out of all these, Eoin has distilled 10 key principles himself:
+
+- Least privilege
+    - Why: Broad privilege allows malicious or accidental access to protected resources
+	- Principle: Limit privileges to the minimum for the context
+    - Tradeoff: It is less convenient, less efficient and more complexity
```
Comment (tomvdbulck) [inline _posts/2016-11-17-JAX-London-2016.md]: and more complexity  => change this to more complex or adds more complexity

File: _posts/2016-11-17-JAX-London-2016.md [hunk 3]
```diff
@@ -0,0 +1,691 @@
...
+<img class="float-image p-image" src="{{ '/img/jax-london-2016/sebastian-meyen.jpeg' | prepend: site.baseurl }}" alt="Sebastian Meyen" style="width: 100px"/>
+
+To start off, [Sebastian Meyen](https://twitter.com/smeyen), program chair of the JAX conferences, gave an introduction to JAX London.
+He mentioned that only half of the attendants of JAX originate from the UK, making it an international conference.
+JAX is all about "openess", we should be celebrating open source and embrace open source thinking as it makes our code smarter.
+Java is one of the most powerful ecosystems out there according to him and it comes with a unique culture.
+
+Sebastian explained that there are different cultures within a company:
+
+* Pioneers: They go into the wild and experiment and although they might fail often, they're looking at what the next thing might be
```
Comment (tomvdbulck) [inline _posts/2016-11-17-JAX-London-2016.md]: what the next thing => what the next big thing 

?? not sure, but the next things sounds odd

File: _posts/2016-11-17-JAX-London-2016.md [hunk 4]
```diff
@@ -0,0 +1,691 @@
...
+
+<img class="float-image p-image" src="{{ '/img/jax-london-2016/james-governor.jpg' | prepend: site.baseurl }}" alt="James Governor" style="width: 100px"/>
+
+[James Governor](https://twitter.com/monkchips) opens up his talk by showing how Java is still high on the Programming Language rankings despite the "Java is dead"-doomspeak every now and then.
+Regarding Java frameworks, Spring and Spring Boot are really crushing it.
+He also mentions that most startups usually start with a new and fancy language but as they mature, a lot of them actually turn into Java shops.
+Examples of this being: Uber, LinkedIn, Netflix, Twitter, Amazon, Etsy, Facebook, Yahoo and Google.
+
+Twitter for example started in Ruby and during the US presidential election in 2012 they migrated to Scala and Java on the JVM for scalability and performance reasons.
+After the migration they managed to sustain peaks of TPS (Tweets Per Second) for hours.
```
Comment (tomvdbulck) [inline _posts/2016-11-17-JAX-London-2016.md]: no # for the TPS?

File: _posts/2016-11-17-JAX-London-2016.md [hunk 5]
```diff
@@ -0,0 +1,691 @@
...
+
+- Bosch: They are doing interesting work with Eclipse Foundation, they know they need to do open source so they make open source contributions
+- Comcast: Everybody hates Comcast in the US, but they are making open source contributions
+
+James mentions a couple of things he finds important:
+
+- Microservices and container based deployment such as Docker and Kubernetes are very hot topics
+- Break down the technical model and teams
+- Continuous integration is very important, there are still people not using it!
+- Make people responsible for their on Quality Assurance
```
Comment (tomvdbulck) [inline _posts/2016-11-17-JAX-London-2016.md]: their own

File: _posts/2016-11-17-JAX-London-2016.md [hunk 6]
```diff
@@ -0,0 +1,691 @@
...
+- Repositories
+- Aggregates
+
+About [Domain-Driven Design aggregates](http://martinfowler.com/bliki/DDD_Aggregate.html):
+
+- Cluster of objects that can be treated as a unit
+- Graph consisting of a root entity and one or more other entities and value objects
+- Typically business entities are aggregates e.g., Customer, Account, Order, Product
+
+Chris talked about the problems you have to deal with when dealing with microservices and that practicing Domain-Driven Design can help you a lot to address these.
```
Comment (tomvdbulck) [inline _posts/2016-11-17-JAX-London-2016.md]: you have to deal with when dealing with ... lot's of with's.

maybe replace to you have to deal with by you have to handle

File: _posts/2016-11-17-JAX-London-2016.md [hunk 7]
```diff
@@ -0,0 +1,691 @@
...
+    - Example: Admins of a submodule should have no access to other module features
+- Trust cautiously
+    - Why: Many security problems are caused by inserting malicious intermediaries in communication paths
+    - Principle: Assume unknown entities are untrusted, have a clear process to establish trust, validate who is connecting
+    - Tradeoff: Operational complexity, reliability, some development overhead
+    - Example: Dont accept untrusted RMI connections, use client certificates, credentials or network controls
+- Simplest solution possible
+    - Why: Security requires understanding of the design, complex design is rarely understood, simplicity allows analysis
+    - Principle: Actively design for simplicity - avoid complex falure modes, implicit behaviour, unnecessary features,…
+    - Tradeoff: Hard decisions on features and sophistication needs serious design effort to be simple
```
Comment (tomvdbulck) [inline _posts/2016-11-17-JAX-London-2016.md]: Hard decisions on features and sophistication needs serious design effort to be simple => Hard decisions on features provided and needs serious design in effort in order to be simple

File: _posts/2016-11-17-JAX-London-2016.md [hunk 8]
```diff
@@ -0,0 +1,691 @@
...
+- Separate responsibilities
+    - Why: Achieve control and accountability, limit the impact of successful attacks, make attacks less attractive
+    - Principle: Separate and compartmentalise responsibilities and privileges
+    - Tradeoff: Development and testing costs, operation complexity, troubleshooting can be more cumbersome
+    - Example: Admins of a submodule should have no access to other module features
+- Trust cautiously
+    - Why: Many security problems are caused by inserting malicious intermediaries in communication paths
+    - Principle: Assume unknown entities are untrusted, have a clear process to establish trust, validate who is connecting
+    - Tradeoff: Operational complexity, reliability, some development overhead
+    - Example: Dont accept untrusted RMI connections, use client certificates, credentials or network controls
```
Comment (Turbots) [inline _posts/2016-11-17-JAX-London-2016.md]: Don't

File: _posts/2016-11-17-JAX-London-2016.md [hunk 9]
```diff
@@ -0,0 +1,691 @@
...
+- Simplest solution possible
+    - Why: Security requires understanding of the design, complex design is rarely understood, simplicity allows analysis
+    - Principle: Actively design for simplicity - avoid complex falure modes, implicit behaviour, unnecessary features,…
```
Comment (Turbots) [inline _posts/2016-11-17-JAX-London-2016.md]: failure

File: _posts/2016-11-17-JAX-London-2016.md [hunk 10]
```diff
@@ -0,0 +1,691 @@
...
+    - Principle: Force changes to security sensitive parameters. Think through failures - Must be secure but recoverable
+    - Tradeoff: Convenience
+    - Example: Don’t allow "SYSTEM/MANAGER" after installation. On failure don’t disable or reset security controls
+- Never rely on obscurity
+    - Why: Hiding things is difficult - someone is going to find them, accidentally if not on purpose
+    - Principle: Assume that the attacker has perfect knowledge, this forces secure system design
+    - Tradeoff: Designing a truly secure system takes time and effort
+    - Example: Assume that an attacker will guess a "port knock" network request sequence or a password encoding
+- Defence in-depth
+    - Why: Systems do get attacked, breaches do happen, mistakes are made - need to minimise impact
```
Comment (tomvdbulck) [inline _posts/2016-11-17-JAX-London-2016.md]: minimise => minimize

File: _posts/2016-11-17-JAX-London-2016.md [hunk 11]
```diff
@@ -0,0 +1,691 @@
...
+- The amount of instances
+	
+Also, in order to encourage squads to register their microservices, if you’re not registering your services in there, you’re not getting any hardware and thus you can’t get your service into production.
+
+In order to maintain their hardware they utilise the Cortana Pool Manager where they have the following information available:
+
+- See hardware available
+- Specify how much you need like how many instances
+
+They currently do a lot of self-hosting but are planning to fully migrate to Google Cloud Platform.
```
Comment (tomvdbulck) [inline _posts/2016-11-17-JAX-London-2016.md]: a link to google cloud platform?

File: _posts/2016-11-17-JAX-London-2016.md [hunk 12]
```diff
@@ -0,0 +1,691 @@
...
+	- New HR systems, management processes, multiple dev locations
+	- Standard Compliance (security, development process)
+
+
+## The Serverless Cloud - Bart Blommaerts 
+<img class="float-image p-image" src="{{ '/img/jax-london-2016/bart-blommaerts.jpg' | prepend: site.baseurl }}" alt="Bart Blommaerts" style="width: 100px"/>
+
+[Bart Blommaerts](https://twitter.com/DaggieBe) starts off explaining that serverless, although as suggested by the name, does not actually mean that you're not using any server.
+What it does mean is that the existence of a server is hidden for you.
+The name might be a bit misplaced and it would be more correct to simply call it Functions as a service (FaaS).
```
Comment (bart-blommaerts) [inline _posts/2016-11-17-JAX-London-2016.md]: Functions as a Service

File: _posts/2016-11-17-JAX-London-2016.md [hunk 13]
```diff
@@ -0,0 +1,691 @@
...
+Serverless or FaaS might actually be the next step in the evolution.
```
Comment (bart-blommaerts) [inline _posts/2016-11-17-JAX-London-2016.md]: evolution of cloud computing.

File: _posts/2016-11-17-JAX-London-2016.md [hunk 14]
```diff
@@ -0,0 +1,691 @@
...
+
+Going serverless will lead to less worries seeing as you no longer have the server management to look out for.
```
Comment (bart-blommaerts) [inline _posts/2016-11-17-JAX-London-2016.md]: the entire server management to take care of?

File: _posts/2016-11-17-JAX-London-2016.md [hunk 15]
```diff
@@ -0,0 +1,691 @@
...
+Security updates, scalability and availability is all taken care off for you by the provider.
```
Comment (bart-blommaerts) [inline _posts/2016-11-17-JAX-London-2016.md]: you, by the provider.

File: _posts/2016-11-17-JAX-London-2016.md [hunk 16]
```diff
@@ -0,0 +1,691 @@
...
+Serverless is stateless, there is no state persisted in between invocations.
+In order to preserve state you can still use a db, file system or cross-application cache.
+
+Bart then listed a couple of the current providers:
+
+- AWS Lambda
+    - Integrates well with all other AWS services
+- IBM Bluemix OpenWhisk
+	- OpenWhisk is open source and available on [Github](https://github.com/openwhisk/openwhisk)
+	- The most mature solution available
```
Comment (bart-blommaerts) [inline _posts/2016-11-17-JAX-London-2016.md]: I didn't say that ... I think Lambda is the most mature solution .. especially taking tooling into consideration.

File: _posts/2016-11-17-JAX-London-2016.md [hunk 17]
```diff
@@ -0,0 +1,691 @@
...
+	- Better UI than AWS and also has a dashboard for monitoring
+- MS Azure Cloud Functions
+	- Mature system
+	- Different paymodel compared to AWS that could be more expensive in the long run
```
Comment (bart-blommaerts) [inline _posts/2016-11-17-JAX-London-2016.md]: payment model?

File: _posts/2016-11-17-JAX-London-2016.md [hunk 18]
```diff
@@ -0,0 +1,691 @@
...
+- Google Cloud Functions
+	- Development by Google started 5 years ago
```
Comment (bart-blommaerts) [inline _posts/2016-11-17-JAX-London-2016.md]: GCF is a new initiative by Google, but it benefits from services Google offered over 5 years ago.

File: _posts/2016-11-17-JAX-London-2016.md [hunk 19]
```diff
@@ -0,0 +1,691 @@
...
+	- Currently reimplementing it
+	- Currently in prime-alpha for which an NDA applies
```
Comment (bart-blommaerts) [inline _posts/2016-11-17-JAX-London-2016.md]: - Currently in private alpha
- Very active community

File: _posts/2016-11-17-JAX-London-2016.md (hunk 16 above)
Comment (yannickdeturck) [inline _posts/2016-11-17-JAX-London-2016.md]: Ok, will leave that out.

File: _posts/2016-11-17-JAX-London-2016.md (hunk 17 above)
Comment (yannickdeturck) [inline _posts/2016-11-17-JAX-London-2016.md]: I believe you mentioned something about MS using a different pay per use system compared to AWS?

Comment (bart-blommaerts) [inline _posts/2016-11-17-JAX-London-2016.md]: Yes. I'd just rephrase from paymodel to payment model.

File: _posts/2016-11-18-JAX-London-2016.md [hunk 20]
```diff
@@ -0,0 +1,691 @@
...
+
+<img class="float-image p-image" src="{{ '/img/jax-london-2016/james-governor.jpg' | prepend: site.baseurl }}" alt="James Governor" style="width: 100px"/>
+
+[James Governor](https://twitter.com/monkchips) opens up his talk by showing how Java is still high on the Programming Language rankings despite the "Java is dead"-doomspeak every now and then.
+Regarding Java frameworks, Spring and Spring Boot are really crushing it.
+He also mentions that most startups usually start with a new and fancy language but as they mature, a lot of them actually turn into Java shops.
+Examples of this being: Uber, LinkedIn, Netflix, Twitter, Amazon, Etsy, Facebook, Yahoo and Google.
+
+Twitter for example started in Ruby and during the US presidential election in 2012 they migrated to Scala and Java on the JVM for scalability and performance reasons.
+After the migration they managed to sustain peaks of TPS (Tweets Per Second) for hours.
```
Comment (yannickdeturck) [inline _posts/2016-11-18-JAX-London-2016.md]: Added a link to https://blog.twitter.com/2012/election-night-2012 (although that's in TPM but they should get the idea).

File: _posts/2016-11-17-JAX-London-2016.md (hunk 10 above)
Comment (yannickdeturck) [inline _posts/2016-11-17-JAX-London-2016.md]: Its British English. :-)

File: _posts/2016-11-17-JAX-London-2016.md [hunk 21]
```diff
@@ -0,0 +1,691 @@
...
+
+System-Z is the service registry in which all services in production get registered in and the following information is available for each microservice:
+
+- Who’s on call for each service
+- When a service is down
+- What the dependencies are for each service
+- The hardware in use
+- The amount of instances
+	
+Also, in order to encourage squads to register their microservices, if you’re not registering your services in there, you’re not getting any hardware and thus you can’t get your service into production.
```
Comment (yannickdeturck) [inline _posts/2016-11-17-JAX-London-2016.md]: I like the original better, I did delete one of the comma's!

Also, in order to encourage squads to register their microservices, if you’re not registering your services in there you’re not getting any hardware and thus you can’t get your service into production.

File: _posts/2016-11-17-JAX-London-2016.md (hunk 11 above)
Comment (yannickdeturck) [inline _posts/2016-11-17-JAX-London-2016.md]: Added.

File: _posts/2016-11-17-JAX-London-2016.md [hunk 22]
```diff
@@ -0,0 +1,691 @@
...
+
+- [Presentation](https://jaxlondon.com/wp-content/uploads/2016/10/Operating-the-Spotify-backend-Niklas-Gustavsson-1.pdf)
+
+## JAX Innovation Awards
+
+The JAX Innovations Awards reward the most innovative contributions in the shape of technology, idea, documentation, industry impact or community engagement.
+
+Most innovative contribution to the Java Ecosystem:
+
+- 1st: Spring Boot
```
Comment (yannickdeturck) [inline _posts/2016-11-17-JAX-London-2016.md]: Added links to all technologies.

File: _posts/2016-11-17-JAX-London-2016.md [hunk 23]
```diff
@@ -0,0 +1,691 @@
...
+- 2nd: Camunda BPM Platform 
+- 3rd: Spark
+
+Innovation in Software delivery & DevOps:
+
+- 1st: Docker
```
Comment (yannickdeturck) [inline _posts/2016-11-17-JAX-London-2016.md]: Added links to all technologies.

**Pull request merged**